### ISSUE TITLE
SegmentationFault with vim-airline

### ISSUE THREAD

**Issue by gerw** [vim/vim#1423]
Recently, I installed https://github.com/vim-airline/vim-airline, but suffered from random vim crashes.

All I have to do is to open `gvim` (using gtk2) and do some resizing of the window. After doing this for a while, I get SegmentationFaults:
````
~/.vim $ gvim vimrc
~/.vim $ Vim: Caught deadly signal SEGV
Vim: Finished.
````
`gdb` is not helpful:
````
~/.vim $ gdb --args gvim vimrc
GNU gdb (Ubuntu 7.11.1-0ubuntu1~16.04) 7.11.1
Copyright (C) 2016 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.  Type "show copying"
and "show warranty" for details.
This GDB was configured as "x86_64-linux-gnu".
Type "show configuration" for configuration details.
For bug reporting instructions, please see:
<http://www.gnu.org/software/gdb/bugs/>.
Find the GDB manual and other documentation resources online at:
<http://www.gnu.org/software/gdb/documentation/>.
For help, type "help".
Type "apropos word" to search for commands related to "word"...
Reading symbols from gvim...done.
(gdb) r
Starting program: /home/gerw/bin/gvim vimrc
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/x86_64-linux-gnu/libthread_db.so.1".
[Inferior 1 (process 14635) exited normally]
(gdb) Vim: Caught deadly signal SEGV
Vim: Finished.
(gdb) bt
No stack.
````
However, valgrind reported some invalid reads: https://gist.github.com/gerw/57e69192b3a9f9cf6dfeb9bb4c8cfbdf

What I have found out:
* it seems that I have to wait the right (very small) amount between to resizings; for valgrind I have to wait a little bit longer
* It seems to only happen on files with trailing spaces and/or mixed indentation (this is displayed by vim-airline in the statusline)

I use the recent vim version from github.

**Comment by dpelle** [comment]
The valgrind stacks that you gave are truncated.  Can you reproduce the bug with "valgrind --num-callers=50 …" so that the stack trace is deeper?

Can you also report which version of Vim you're using? Try to compile the latest Vim from github if possible, in case this bug has already been fixed. In particular, patch 8.0.0020 made the regexp re-entrant.  It might be the the fix to this bug (not sure).

**Comment by gerw** [comment]
I reran with `valgrind --num-callers=50`, see the updated gist. However, the stack is still truncated. Is this now usefull, or should I rerun with an even higher number?

I use the latest version from github.

**Comment by chrisbra** [comment]
please show the output of `:version`. Also since you seem to be able to reproduce this crash, please provide a gdb backtrace.

**Comment by gerw** [comment]
I have added the output of `gvim --version` to the gist. I am not able to provide a gdb backtrace, since gdb says `No stack.` (see my first post). If there is anything I can do with gdb to produce a backtrace, please let me know.

**Comment by dpelle** [comment]
I installed vim-airline and I after a few trials, I found a crash
using gvim-8.0.264 (gtk2 GUI) on Linux 86_64, with a asan build.

My stack share many similarities with the stack of bug submitter, even if
they are different.

I can reproduce by doing:

$ vim -f -g -c 'help' 2> log

I then reduce the window size and gvim crashes often (not all the time).
Here is asan report:
```
=================================================================
==9702==ERROR: AddressSanitizer: heap-use-after-free on address 0x602000507951 at pc 0x000000d1a696 bp 0x7fff352762f0 sp 0x7fff352762e8
READ of size 1 at 0x602000507951 thread T0
    #0 0xd1a695 in utf_ptr2char /home/pel/sb/vim/src/mbyte.c:1761
    #1 0xfb4fb2 in nfa_regmatch /home/pel/sb/vim/src/./regexp_nfa.c:5638
    #2 0xfb18e2 in nfa_regtry /home/pel/sb/vim/src/./regexp_nfa.c:6990
    #3 0xfae9cc in nfa_regexec_both /home/pel/sb/vim/src/./regexp_nfa.c:7182
    #4 0xfad3e0 in nfa_regexec_multi /home/pel/sb/vim/src/./regexp_nfa.c:7392
    #5 0xfaa88c in vim_regexec_multi /home/pel/sb/vim/src/regexp.c:8318
    #6 0x1386a00 in syn_regexec /home/pel/sb/vim/src/syntax.c:3308
    #7 0x13109e7 in syn_current_attr /home/pel/sb/vim/src/syntax.c:2117
    #8 0x130d21f in get_syntax_attr /home/pel/sb/vim/src/syntax.c:1869
    #9 0x10e1159 in win_line /home/pel/sb/vim/src/screen.c:4419
    #10 0x10c2b05 in update_single_line /home/pel/sb/vim/src/screen.c:850
    #11 0x16e2897 in main_loop /home/pel/sb/vim/src/main.c:1258
    #12 0x16dbefa in vim_main2 /home/pel/sb/vim/src/main.c:877
    #13 0x16d0af7 in main /home/pel/sb/vim/src/main.c:415
    #14 0x7ff79d48bf44 in __libc_start_main /build/eglibc-oGUzwX/eglibc-2.19/csu/libc-start.c:287
    #15 0x46adb6 in _start ??:?

0x602000507951 is located 1 bytes inside of 2-byte region [0x602000507950,0x602000507952)
freed by thread T0 here:
    #0 0x4f1aa2 in free ??:?
    #1 0xca2c36 in vim_free /home/pel/sb/vim/src/misc2.c:1793
    #2 0x6716d5 in clear_tv /home/pel/sb/vim/src/eval.c:7082
    #3 0x6ded6c in eval4 /home/pel/sb/vim/src/eval.c:3779
    #4 0x6d83f5 in eval3 /home/pel/sb/vim/src/eval.c:3433
    #5 0x6929b5 in eval2 /home/pel/sb/vim/src/eval.c:3365
    #6 0x672a96 in eval1 /home/pel/sb/vim/src/eval.c:3293
    #7 0x670751 in eval0 /home/pel/sb/vim/src/eval.c:3253
    #8 0x66f47f in eval_to_bool /home/pel/sb/vim/src/eval.c:679
    #9 0x933c65 in ex_else /home/pel/sb/vim/src/ex_eval.c:1013
    #10 0x893ad2 in do_one_cmd /home/pel/sb/vim/src/ex_docmd.c:2981
    #11 0x87c292 in do_cmdline /home/pel/sb/vim/src/ex_docmd.c:1120
    #12 0x1483fee in call_user_func /home/pel/sb/vim/src/userfunc.c:893
    #13 0x147b6ae in call_func /home/pel/sb/vim/src/userfunc.c:1353
    #14 0x1478327 in get_func_tv /home/pel/sb/vim/src/userfunc.c:455
    #15 0x6e6b10 in eval7 /home/pel/sb/vim/src/eval.c:4364
    #16 0x6e1b7f in eval6 /home/pel/sb/vim/src/eval.c:4001
    #17 0x6df544 in eval5 /home/pel/sb/vim/src/eval.c:3817
    #18 0x6d95e2 in eval4 /home/pel/sb/vim/src/eval.c:3516
    #19 0x6d83f5 in eval3 /home/pel/sb/vim/src/eval.c:3433
    #20 0x6929b5 in eval2 /home/pel/sb/vim/src/eval.c:3365
    #21 0x672a96 in eval1 /home/pel/sb/vim/src/eval.c:3293
    #22 0x670751 in eval0 /home/pel/sb/vim/src/eval.c:3253
    #23 0x673c3f in eval_to_string /home/pel/sb/vim/src/eval.c:757
    #24 0x674605 in eval_to_string_safe /home/pel/sb/vim/src/eval.c:805
    #25 0x5542f0 in build_stl_str_hl /home/pel/sb/vim/src/buffer.c:4165
    #26 0x1139db1 in win_redr_custom /home/pel/sb/vim/src/screen.c:7050
    #27 0x110fc74 in redraw_custom_statusline /home/pel/sb/vim/src/screen.c:6833
    #28 0x10bb74b in win_redr_status /home/pel/sb/vim/src/screen.c:6704
    #29 0x108f43b in update_screen /home/pel/sb/vim/src/screen.c:719

previously allocated by thread T0 here:
    #0 0x4f1d82 in __interceptor_malloc ??:?
    #1 0xca00f0 in lalloc /home/pel/sb/vim/src/misc2.c:942
    #2 0xc9fd7b in alloc /home/pel/sb/vim/src/misc2.c:840
    #3 0xca2e6c in vim_strsave /home/pel/sb/vim/src/misc2.c:1285
    #4 0x6ae762 in copy_tv /home/pel/sb/vim/src/eval.c:7970
    #5 0x6ad9ba in get_var_tv /home/pel/sb/vim/src/eval.c:6857
    #6 0x6e70dc in eval7 /home/pel/sb/vim/src/eval.c:4389
    #7 0x6e1b7f in eval6 /home/pel/sb/vim/src/eval.c:4001
    #8 0x6df544 in eval5 /home/pel/sb/vim/src/eval.c:3817
    #9 0x6d95e2 in eval4 /home/pel/sb/vim/src/eval.c:3516
    #10 0x6d83f5 in eval3 /home/pel/sb/vim/src/eval.c:3433
    #11 0x6929b5 in eval2 /home/pel/sb/vim/src/eval.c:3365
    #12 0x672a96 in eval1 /home/pel/sb/vim/src/eval.c:3293
    #13 0x670751 in eval0 /home/pel/sb/vim/src/eval.c:3253
    #14 0x66f47f in eval_to_bool /home/pel/sb/vim/src/eval.c:679
    #15 0x933c65 in ex_else /home/pel/sb/vim/src/ex_eval.c:1013
    #16 0x893ad2 in do_one_cmd /home/pel/sb/vim/src/ex_docmd.c:2981
    #17 0x87c292 in do_cmdline /home/pel/sb/vim/src/ex_docmd.c:1120
    #18 0x1483fee in call_user_func /home/pel/sb/vim/src/userfunc.c:893
    #19 0x147b6ae in call_func /home/pel/sb/vim/src/userfunc.c:1353
    #20 0x1478327 in get_func_tv /home/pel/sb/vim/src/userfunc.c:455
    #21 0x6e6b10 in eval7 /home/pel/sb/vim/src/eval.c:4364
    #22 0x6e1b7f in eval6 /home/pel/sb/vim/src/eval.c:4001
    #23 0x6df544 in eval5 /home/pel/sb/vim/src/eval.c:3817
    #24 0x6d95e2 in eval4 /home/pel/sb/vim/src/eval.c:3516
    #25 0x6d83f5 in eval3 /home/pel/sb/vim/src/eval.c:3433
    #26 0x6929b5 in eval2 /home/pel/sb/vim/src/eval.c:3365
    #27 0x672a96 in eval1 /home/pel/sb/vim/src/eval.c:3293
    #28 0x670751 in eval0 /home/pel/sb/vim/src/eval.c:3253
    #29 0x673c3f in eval_to_string /home/pel/sb/vim/src/eval.c:757

Shadow bytes around the buggy address:
  0x0c0480098ed0: fa fa fd fd fa fa fd fd fa fa fd fa fa fa fd fa
  0x0c0480098ee0: fa fa fd fa fa fa fd fd fa fa fd fd fa fa fd fa
  0x0c0480098ef0: fa fa fd fa fa fa fd fa fa fa 07 fa fa fa fd fa
  0x0c0480098f00: fa fa fd fa fa fa fd fd fa fa fd fa fa fa fd fa
  0x0c0480098f10: fa fa fd fa fa fa fd fa fa fa fd fa fa fa fd fa
=>0x0c0480098f20: fa fa fd fa fa fa fd fa fa fa[fd]fa fa fa fd fa
  0x0c0480098f30: fa fa fd fa fa fa fd fa fa fa fd fd fa fa fd fa
  0x0c0480098f40: fa fa fd fa fa fa fd fa fa fa fd fa fa fa fd fa
  0x0c0480098f50: fa fa fd fa fa fa fd fa fa fa fd fa fa fa fd fa
  0x0c0480098f60: fa fa fd fd fa fa fd fa fa fa fd fd fa fa fd fa
  0x0c0480098f70: fa fa fd fa fa fa fd fd fa fa fd fa fa fa fd fd
Shadow byte legend (one shadow byte represents 8 application bytes):
  Addressable:           00
  Partially addressable: 01 02 03 04 05 06 07
  Heap left redzone:       fa
  Heap right redzone:      fb
  Freed heap region:       fd
  Stack left redzone:      f1
  Stack mid redzone:       f2
  Stack right redzone:     f3
  Stack partial redzone:   f4
  Stack after return:      f5
  Stack use after scope:   f8
  Global redzone:          f9
  Global init order:       f6
  Poisoned by user:        f7
  Container overflow:      fc
  Array cookie:            ac
  Intra object redzone:    bb
  ASan internal:           fe
  Left alloca redzone:     ca
  Right alloca redzone:    cb
==9702==ABORTING
```

**Comment by gerw** [comment]
Using your invocation, vim also reports about a violated assertion:
````
gvim: malloc.c:2392: sysmalloc: Assertion `(old_top == initial_top (av) && old_size == 0) || ((unsigned long) (old_size) >= MINSIZE && prev_inuse (old_top) && ((unsigned long) old_end & (pagesize - 1)) == 0)' failed.
````

**Comment by dpelle** [comment]
Here is a complete stack with valgrind using gvim 8.0.271 gtk2:

$ valgrind --num-callers=120 ./vim -f -g -c help 2> vg.log

... and then resizing the window a couple times gives this error:

```
==11663== Memcheck, a memory error detector
==11663== Copyright (C) 2002-2015, and GNU GPL'd, by Julian Seward et al.
==11663== Using Valgrind-3.12.0.SVN and LibVEX; rerun with -h for copyright info
==11663== Command: ./vim -f -g -c help
==11663== 
==11663== Invalid read of size 1
==11663==    at 0x53458C: utf_ptr2char (mbyte.c:1761)
==11663==    by 0x5957CF: nfa_regmatch (regexp_nfa.c:5638)
==11663==    by 0x595015: nfa_regtry (regexp_nfa.c:6990)
==11663==    by 0x594A6B: nfa_regexec_both (regexp_nfa.c:7182)
==11663==    by 0x5946CE: nfa_regexec_multi (regexp_nfa.c:7392)
==11663==    by 0x594078: vim_regexec_multi (regexp.c:8318)
==11663==    by 0x627B41: syn_regexec (syntax.c:3308)
==11663==    by 0x616F7B: syn_current_attr (syntax.c:2117)
==11663==    by 0x6165E1: get_syntax_attr (syntax.c:1869)
==11663==    by 0x5CB3FE: win_line (screen.c:4419)
==11663==    by 0x5C7665: update_single_line (screen.c:850)
==11663==    by 0x6A2A75: main_loop (main.c:1258)
==11663==    by 0x6A0E78: vim_main2 (main.c:877)
==11663==    by 0x69E54E: main (main.c:415)
==11663==  Address 0xf6c8ac1 is 1 bytes inside a block of size 2 free'd
==11663==    at 0x4C2BCEF: free (vg_replace_malloc.c:530)
==11663==    by 0x525DC3: vim_free (misc2.c:1793)
==11663==    by 0x443A5E: clear_tv (eval.c:7082)
==11663==    by 0x4534B5: eval4 (eval.c:3779)
==11663==    by 0x452539: eval3 (eval.c:3433)
==11663==    by 0x448729: eval2 (eval.c:3365)
==11663==    by 0x443CA2: eval1 (eval.c:3293)
==11663==    by 0x443793: eval0 (eval.c:3253)
==11663==    by 0x443480: eval_to_bool (eval.c:679)
==11663==    by 0x4A9B77: ex_else (ex_eval.c:1013)
==11663==    by 0x4929F9: do_one_cmd (ex_docmd.c:2981)
==11663==    by 0x48E053: do_cmdline (ex_docmd.c:1120)
==11663==    by 0x64B322: call_user_func (userfunc.c:893)
==11663==    by 0x649EF4: call_func (userfunc.c:1353)
==11663==    by 0x6497CB: get_func_tv (userfunc.c:455)
==11663==    by 0x4543C4: eval7 (eval.c:4364)
==11663==    by 0x4539AE: eval6 (eval.c:4001)
==11663==    by 0x453524: eval5 (eval.c:3817)
==11663==    by 0x45275A: eval4 (eval.c:3516)
==11663==    by 0x452539: eval3 (eval.c:3433)
==11663==    by 0x448729: eval2 (eval.c:3365)
==11663==    by 0x443CA2: eval1 (eval.c:3293)
==11663==    by 0x443793: eval0 (eval.c:3253)
==11663==    by 0x443F1E: eval_to_string (eval.c:757)
==11663==    by 0x4440CD: eval_to_string_safe (eval.c:805)
==11663==    by 0x41A106: build_stl_str_hl (buffer.c:4165)
==11663==    by 0x5D5C15: win_redr_custom (screen.c:7050)
==11663==    by 0x5D0C0C: redraw_custom_statusline (screen.c:6833)
==11663==    by 0x5C658E: win_redr_status (screen.c:6704)
==11663==    by 0x5C1516: update_screen (screen.c:719)
==11663==    by 0x634825: set_shellsize (term.c:3074)
==11663==    by 0x635BB0: shell_resized (term.c:2938)
==11663==    by 0x66452B: gui_resize_shell (gui.c:1484)
==11663==    by 0x675F5A: form_configure_event (gui_gtk_x11.c:4534)
==11663==    by 0x4F63814: ??? (in /usr/lib/x86_64-linux-gnu/libgtk-x11-2.0.so.0.2400.23)
==11663==    by 0x5BA43B7: g_closure_invoke (in /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0.4002.0)
==11663==    by 0x5BB5D3C: ??? (in /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0.4002.0)
==11663==    by 0x5BBD6F8: g_signal_emit_valist (in /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0.4002.0)
==11663==    by 0x5BBDCE1: g_signal_emit (in /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0.4002.0)
==11663==    by 0x5073683: ??? (in /usr/lib/x86_64-linux-gnu/libgtk-x11-2.0.so.0.2400.23)
==11663==    by 0x4F6257A: gtk_main_do_event (in /usr/lib/x86_64-linux-gnu/libgtk-x11-2.0.so.0.2400.23)
==11663==    by 0x67DC5F: gtk_form_send_configure (gui_gtk_f.c:1203)
==11663==    by 0x67D596: gtk_form_size_allocate (gui_gtk_f.c:674)
==11663==    by 0x5BA720F: g_cclosure_marshal_VOID__BOXEDv (in /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0.4002.0)
==11663==    by 0x5BA455E: ??? (in /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0.4002.0)
==11663==    by 0x5BBD087: g_signal_emit_valist (in /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0.4002.0)
==11663==    by 0x5BBDCE1: g_signal_emit (in /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0.4002.0)
==11663==    by 0x5077509: gtk_widget_size_allocate (in /usr/lib/x86_64-linux-gnu/libgtk-x11-2.0.so.0.2400.23)
==11663==    by 0x4EB90F7: ??? (in /usr/lib/x86_64-linux-gnu/libgtk-x11-2.0.so.0.2400.23)
==11663==    by 0x5BA720F: g_cclosure_marshal_VOID__BOXEDv (in /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0.4002.0)
==11663==    by 0x5BA455E: ??? (in /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0.4002.0)
==11663==    by 0x5BBD087: g_signal_emit_valist (in /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0.4002.0)
==11663==    by 0x5BBDCE1: g_signal_emit (in /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0.4002.0)
==11663==    by 0x5077509: gtk_widget_size_allocate (in /usr/lib/x86_64-linux-gnu/libgtk-x11-2.0.so.0.2400.23)
==11663==    by 0x5080EA9: ??? (in /usr/lib/x86_64-linux-gnu/libgtk-x11-2.0.so.0.2400.23)
==11663==    by 0x5BA720F: g_cclosure_marshal_VOID__BOXEDv (in /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0.4002.0)
==11663==    by 0x5BA45E6: ??? (in /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0.4002.0)
==11663==    by 0x5BBD087: g_signal_emit_valist (in /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0.4002.0)
==11663==    by 0x5BBDCE1: g_signal_emit (in /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0.4002.0)
==11663==    by 0x5077509: gtk_widget_size_allocate (in /usr/lib/x86_64-linux-gnu/libgtk-x11-2.0.so.0.2400.23)
==11663==    by 0x508200B: ??? (in /usr/lib/x86_64-linux-gnu/libgtk-x11-2.0.so.0.2400.23)
==11663==    by 0x5BA45E6: ??? (in /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0.4002.0)
==11663==    by 0x5BBD087: g_signal_emit_valist (in /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0.4002.0)
==11663==    by 0x5BBDCE1: g_signal_emit (in /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0.4002.0)
==11663==    by 0x4EEAE1F: ??? (in /usr/lib/x86_64-linux-gnu/libgtk-x11-2.0.so.0.2400.23)
==11663==    by 0x5490CE6: ??? (in /usr/lib/x86_64-linux-gnu/libgdk-x11-2.0.so.0.2400.23)
==11663==    by 0x5E2DCE4: g_main_context_dispatch (in /lib/x86_64-linux-gnu/libglib-2.0.so.0.4002.0)
==11663==    by 0x5E2E047: ??? (in /lib/x86_64-linux-gnu/libglib-2.0.so.0.4002.0)
==11663==    by 0x5E2E0EB: g_main_context_iteration (in /lib/x86_64-linux-gnu/libglib-2.0.so.0.4002.0)
==11663==    by 0x671198: gui_mch_update (gui_gtk_x11.c:6576)
==11663==    by 0x63B610: ui_breakcheck_force (ui.c:368)
==11663==    by 0x63B5EA: ui_breakcheck (ui.c:356)
==11663==    by 0x520511: line_breakcheck (misc1.c:9554)
==11663==    by 0x598D71: nfa_regmatch (regexp_nfa.c:6908)
==11663==    by 0x595015: nfa_regtry (regexp_nfa.c:6990)
==11663==    by 0x594A6B: nfa_regexec_both (regexp_nfa.c:7182)
==11663==    by 0x5946CE: nfa_regexec_multi (regexp_nfa.c:7392)
==11663==    by 0x594078: vim_regexec_multi (regexp.c:8318)
==11663==    by 0x627B41: syn_regexec (syntax.c:3308)
==11663==    by 0x616F7B: syn_current_attr (syntax.c:2117)
==11663==    by 0x6165E1: get_syntax_attr (syntax.c:1869)
==11663==    by 0x5CB3FE: win_line (screen.c:4419)
==11663==    by 0x5C7665: update_single_line (screen.c:850)
==11663==    by 0x6A2A75: main_loop (main.c:1258)
==11663==    by 0x6A0E78: vim_main2 (main.c:877)
==11663==    by 0x69E54E: main (main.c:415)
==11663==  Block was alloc'd at
==11663==    at 0x4C2ABF5: malloc (vg_replace_malloc.c:299)
==11663==    by 0x524DFD: lalloc (misc2.c:942)
==11663==    by 0x524D97: alloc (misc2.c:840)
==11663==    by 0x525127: vim_strsave (misc2.c:1285)
==11663==    by 0x44C282: copy_tv (eval.c:7970)
==11663==    by 0x44C141: get_var_tv (eval.c:6857)
==11663==    by 0x454475: eval7 (eval.c:4389)
==11663==    by 0x4539AE: eval6 (eval.c:4001)
==11663==    by 0x453524: eval5 (eval.c:3817)
==11663==    by 0x45275A: eval4 (eval.c:3516)
==11663==    by 0x452539: eval3 (eval.c:3433)
==11663==    by 0x448729: eval2 (eval.c:3365)
==11663==    by 0x443CA2: eval1 (eval.c:3293)
==11663==    by 0x443793: eval0 (eval.c:3253)
==11663==    by 0x443480: eval_to_bool (eval.c:679)
==11663==    by 0x4A9B77: ex_else (ex_eval.c:1013)
==11663==    by 0x4929F9: do_one_cmd (ex_docmd.c:2981)
==11663==    by 0x48E053: do_cmdline (ex_docmd.c:1120)
==11663==    by 0x64B322: call_user_func (userfunc.c:893)
==11663==    by 0x649EF4: call_func (userfunc.c:1353)
==11663==    by 0x6497CB: get_func_tv (userfunc.c:455)
==11663==    by 0x4543C4: eval7 (eval.c:4364)
==11663==    by 0x4539AE: eval6 (eval.c:4001)
==11663==    by 0x453524: eval5 (eval.c:3817)
==11663==    by 0x45275A: eval4 (eval.c:3516)
==11663==    by 0x452539: eval3 (eval.c:3433)
==11663==    by 0x448729: eval2 (eval.c:3365)
==11663==    by 0x443CA2: eval1 (eval.c:3293)
==11663==    by 0x443793: eval0 (eval.c:3253)
==11663==    by 0x443F1E: eval_to_string (eval.c:757)
==11663==    by 0x4440CD: eval_to_string_safe (eval.c:805)
==11663==    by 0x41A106: build_stl_str_hl (buffer.c:4165)
==11663==    by 0x5D5C15: win_redr_custom (screen.c:7050)
==11663==    by 0x5D0C0C: redraw_custom_statusline (screen.c:6833)
==11663==    by 0x5C658E: win_redr_status (screen.c:6704)
==11663==    by 0x5C1516: update_screen (screen.c:719)
==11663==    by 0x634825: set_shellsize (term.c:3074)
==11663==    by 0x635BB0: shell_resized (term.c:2938)
==11663==    by 0x66452B: gui_resize_shell (gui.c:1484)
==11663==    by 0x675F5A: form_configure_event (gui_gtk_x11.c:4534)
==11663==    by 0x4F63814: ??? (in /usr/lib/x86_64-linux-gnu/libgtk-x11-2.0.so.0.2400.23)
==11663==    by 0x5BA43B7: g_closure_invoke (in /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0.4002.0)
==11663==    by 0x5BB5D3C: ??? (in /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0.4002.0)
==11663==    by 0x5BBD6F8: g_signal_emit_valist (in /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0.4002.0)
==11663==    by 0x5BBDCE1: g_signal_emit (in /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0.4002.0)
==11663==    by 0x5073683: ??? (in /usr/lib/x86_64-linux-gnu/libgtk-x11-2.0.so.0.2400.23)
==11663==    by 0x4F6257A: gtk_main_do_event (in /usr/lib/x86_64-linux-gnu/libgtk-x11-2.0.so.0.2400.23)
==11663==    by 0x67DC5F: gtk_form_send_configure (gui_gtk_f.c:1203)
==11663==    by 0x67D596: gtk_form_size_allocate (gui_gtk_f.c:674)
==11663==    by 0x5BA720F: g_cclosure_marshal_VOID__BOXEDv (in /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0.4002.0)
==11663==    by 0x5BA455E: ??? (in /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0.4002.0)
==11663==    by 0x5BBD087: g_signal_emit_valist (in /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0.4002.0)
==11663==    by 0x5BBDCE1: g_signal_emit (in /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0.4002.0)
==11663==    by 0x5077509: gtk_widget_size_allocate (in /usr/lib/x86_64-linux-gnu/libgtk-x11-2.0.so.0.2400.23)
==11663==    by 0x4EB90F7: ??? (in /usr/lib/x86_64-linux-gnu/libgtk-x11-2.0.so.0.2400.23)
==11663==    by 0x5BA720F: g_cclosure_marshal_VOID__BOXEDv (in /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0.4002.0)
==11663==    by 0x5BA455E: ??? (in /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0.4002.0)
==11663==    by 0x5BBD087: g_signal_emit_valist (in /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0.4002.0)
==11663==    by 0x5BBDCE1: g_signal_emit (in /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0.4002.0)
==11663==    by 0x5077509: gtk_widget_size_allocate (in /usr/lib/x86_64-linux-gnu/libgtk-x11-2.0.so.0.2400.23)
==11663==    by 0x5080EA9: ??? (in /usr/lib/x86_64-linux-gnu/libgtk-x11-2.0.so.0.2400.23)
==11663==    by 0x5BA720F: g_cclosure_marshal_VOID__BOXEDv (in /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0.4002.0)
==11663==    by 0x5BA45E6: ??? (in /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0.4002.0)
==11663==    by 0x5BBD087: g_signal_emit_valist (in /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0.4002.0)
==11663==    by 0x5BBDCE1: g_signal_emit (in /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0.4002.0)
==11663==    by 0x5077509: gtk_widget_size_allocate (in /usr/lib/x86_64-linux-gnu/libgtk-x11-2.0.so.0.2400.23)
==11663==    by 0x508200B: ??? (in /usr/lib/x86_64-linux-gnu/libgtk-x11-2.0.so.0.2400.23)
==11663==    by 0x5BA45E6: ??? (in /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0.4002.0)
==11663==    by 0x5BBD087: g_signal_emit_valist (in /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0.4002.0)
==11663==    by 0x5BBDCE1: g_signal_emit (in /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0.4002.0)
==11663==    by 0x4EEAE1F: ??? (in /usr/lib/x86_64-linux-gnu/libgtk-x11-2.0.so.0.2400.23)
==11663==    by 0x5490CE6: ??? (in /usr/lib/x86_64-linux-gnu/libgdk-x11-2.0.so.0.2400.23)
==11663==    by 0x5E2DCE4: g_main_context_dispatch (in /lib/x86_64-linux-gnu/libglib-2.0.so.0.4002.0)
==11663==    by 0x5E2E047: ??? (in /lib/x86_64-linux-gnu/libglib-2.0.so.0.4002.0)
==11663==    by 0x5E2E0EB: g_main_context_iteration (in /lib/x86_64-linux-gnu/libglib-2.0.so.0.4002.0)
==11663==    by 0x671198: gui_mch_update (gui_gtk_x11.c:6576)
==11663==    by 0x63B610: ui_breakcheck_force (ui.c:368)
==11663==    by 0x63B5EA: ui_breakcheck (ui.c:356)
==11663==    by 0x520511: line_breakcheck (misc1.c:9554)
==11663==    by 0x598D71: nfa_regmatch (regexp_nfa.c:6908)
==11663==    by 0x595015: nfa_regtry (regexp_nfa.c:6990)
==11663==    by 0x594A6B: nfa_regexec_both (regexp_nfa.c:7182)
==11663==    by 0x5946CE: nfa_regexec_multi (regexp_nfa.c:7392)
==11663==    by 0x594078: vim_regexec_multi (regexp.c:8318)
==11663==    by 0x627B41: syn_regexec (syntax.c:3308)
==11663==    by 0x616F7B: syn_current_attr (syntax.c:2117)
==11663==    by 0x6165E1: get_syntax_attr (syntax.c:1869)
==11663==    by 0x5CB3FE: win_line (screen.c:4419)
==11663==    by 0x5C7665: update_single_line (screen.c:850)
==11663==    by 0x6A2A75: main_loop (main.c:1258)
==11663==    by 0x6A0E78: vim_main2 (main.c:877)
==11663==    by 0x69E54E: main (main.c:415)
...snip more errors...
```

So at regexp_nfa.c:6908, line_breakcheck() is called, and inside
that call the shell_resized is handled, which frees some memory
and somehow this messed up the regexp engine which later uses
free memorry.

I'm not sure how to properly fix this, but with the following
workaround, it never crashes:

```
diff --git a/src/regexp_nfa.c b/src/regexp_nfa.c
index de22dcb..a561234 100644
--- a/src/regexp_nfa.c
+++ b/src/regexp_nfa.c
@@ -6905,7 +6905,7 @@ nextchar:
            break;
 
        /* Allow interrupting with CTRL-C. */
-       line_breakcheck();
+       /* line_breakcheck(); */
        if (got_int)
            break;
 #ifdef FEAT_RELTIME
```

Can bug submitter also verify that this patch avoids the crash?

A better patch may be needed, but it hopefully helps to diagnose the issue.

**Comment by chrisbra** [comment]
Hm, update_single_line seems to be related to conceal handling. Was your cursor in a concealed document? I ask, because I cannot reproduce this issue, although I keep my cursor in a help document and resize it.
Does this patch fix it?
```patch
diff --git a/src/screen.c b/src/screen.c
index e37bf4838..b758751ea 100644
--- a/src/screen.c
+++ b/src/screen.c
@@ -824,12 +824,13 @@ update_single_line(win_T *wp, linenr_T lnum)
     int                j;

     /* Don't do anything if the screen structures are (not yet) valid. */
-    if (!screen_valid(TRUE))
+    if (!screen_valid(TRUE) || updating_screen)
        return;

     if (lnum >= wp->w_topline && lnum < wp->w_botline
                                 && foldedCount(wp, lnum, &win_foldinfo) == 0)
     {
+       updating_screen = TRUE;
 # ifdef FEAT_GUI
        /* Remove the cursor before starting to do anything, because scrolling
         * may make it difficult to redraw the text under it. */
@@ -862,6 +863,7 @@ update_single_line(win_T *wp, linenr_T lnum)
            out_flush();        /* required before updating the cursor */
            gui_update_cursor(FALSE, FALSE);
        }
+       updating_screen = FALSE;
 # endif
     }
     need_cursor_line_redraw = FALSE;
```

**Comment by nuko8** [comment]
> Hm, update_single_line seems to be related to conceal handling. 

That reminds me of [an outstanding issue](https://groups.google.com/forum/#!msg/vim_dev/dOrbhGPQXrM/xI_YsJNiBgAJ;context-place=forum/vim_dev) where the OP of that thread reported that a combination of a certain window manager and a certain plugin had a strong tendency to induce segfault when gvim was being resized.

> I ask, because I cannot reproduce this issue, although I keep my cursor in a help document and resize it.

In the thread quoted above, the OP also reported that the plugin was indentline, and that, when he disabled it, gvim stopped crashing. This seemingly matches the description given by the OP of this thread:

> It seems to only happen on files with trailing spaces and/or mixed indentation...

So, instead of using vim-airline, trying that plugin might come in handy to reproduce the issue.

**Comment by dpelle** [comment]
@chrisbra wrote:

> Hm, update_single_line seems to be related to conceal handling.

Yes, that was also my guess, which is why I tried a help page to
reproduce the bug (links use concealled bars).

In fact, it crashes when doing...

$ vim -f -g -c help -c 'set conceallevel=2'   (then reduce window size -> CRASH)

But it does not crash when doing:

$ vim -f -g -c help -c 'set conceallevel=0'   (then reduce window size -> no crash)

... which confirms that conceal is trigger the bug.

> Was your cursor in a concealed document? I ask, because I cannot reproduce
> this issue, although I keep my cursor in a help document and resize it.

Yes, when I run "vim -f -g -c help", the cursor is on the first char of the
first line of the help window i.e.  on the H of the "Help.txt" link.
But it does not seem to matter anyway, it also crash if I move the cursor
elsewhere before resizing.

I'm not sure why you can't reproduce. There might be something needed
in my ~/.vimr.  When I find time, I will try to come up with the minimalistic
~/.vimrc to reproduce it.

I used the GTK2 vim GUI with vim huge. Here is the full version description:

```
$ ./vim --version
VIM - Vi IMproved 8.0 (2016 Sep 12, compiled Jan 30 2017 21:42:45)
Included patches: 1-271
Compiled by pel@pel-laptop
Huge version with GTK2 GUI.  Features included (+) or not (-):
+acl             +file_in_path    +mouse_sgr       +tag_old_static
+arabic          +find_in_path    -mouse_sysmouse  -tag_any_white
+autocmd         +float           +mouse_urxvt     -tcl
+balloon_eval    +folding         +mouse_xterm     +termguicolors
+browse          -footer          +multi_byte      +terminfo
++builtin_terms  +fork()          +multi_lang      +termresponse
+byte_offset     +gettext         -mzscheme        +textobjects
+channel         -hangul_input    +netbeans_intg   +timers
+cindent         +iconv           +num64           +title
+clientserver    +insert_expand   +packages        +toolbar
+clipboard       +job             +path_extra      +user_commands
+cmdline_compl   +jumplist        -perl            +vertsplit
+cmdline_hist    +keymap          +persistent_undo +virtualedit
+cmdline_info    +lambda          +postscript      +visual
+comments        +langmap         +printer         +visualextra
+conceal         +libcall         +profile         +viminfo
+cryptv          +linebreak       -python          +vreplace
+cscope          +lispindent      -python3         +wildignore
+cursorbind      +listcmds        +quickfix        +wildmenu
+cursorshape     +localmap        +reltime         +windows
+dialog_con_gui  +lua             +rightleft       +writebackup
+diff            +menu            -ruby            +X11
+digraphs        +mksession       +scrollbind      -xfontset
+dnd             +modify_fname    +signs           +xim
-ebcdic          +mouse           +smartindent     +xpm
+emacs_tags      +mouseshape      +startuptime     +xsmp_interact
+eval            +mouse_dec       +statusline      +xterm_clipboard
+ex_extra        +mouse_gpm       -sun_workshop    -xterm_save
+extra_search    -mouse_jsbterm   +syntax          
+farsi           +mouse_netterm   +tag_binary      
   system vimrc file: "$VIM/vimrc"
     user vimrc file: "$HOME/.vimrc"
 2nd user vimrc file: "~/.vim/vimrc"
      user exrc file: "$HOME/.exrc"
  system gvimrc file: "$VIM/gvimrc"
    user gvimrc file: "$HOME/.gvimrc"
2nd user gvimrc file: "~/.vim/gvimrc"
       defaults file: "$VIMRUNTIME/defaults.vim"
    system menu file: "$VIMRUNTIME/menu.vim"
  fall-back for $VIM: "/usr/local/share/vim"
Compilation: clang-3.6 -c -I. -Iproto -DHAVE_CONFIG_H -DFEAT_GUI_GTK  -pthread -I/usr/include/gtk-2.0 -I/usr/lib/x86_64-linux-gnu/gtk-2.0/include -I/usr/include/atk-1.0 -I/usr/include/cairo -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/pango-1.0 -I/usr/include/gio-unix-2.0/ -I/usr/include/freetype2 -I/usr/include/glib-2.0 -I/usr/lib/x86_64-linux-gnu/glib-2.0/include -I/usr/include/pixman-1 -I/usr/include/libpng12 -I/usr/include/harfbuzz     -g -O0 -fno-strength-reduce -Wall -Wshadow -Wmissing-prototypes  -g -O0 -fsanitize=address -fno-omit-frame-pointer  -DABORT_ON_INTERNAL_ERROR   
Linking: clang-3.6   -L/usr/local/lib -Wl,--as-needed -o vim   -lgtk-x11-2.0 -lgdk-x11-2.0 -latk-1.0 -lgio-2.0 -lpangoft2-1.0 -lpangocairo-1.0 -lgdk_pixbuf-2.0 -lcairo -lpango-1.0 -lfontconfig -lgobject-2.0 -lglib-2.0 -lfreetype   -lSM -lICE -lXpm -lXt -lX11 -lXdmcp -lSM -lICE  -lm -ltinfo -lnsl  -lselinux -lacl -lattr -lgpm -ldl  -L/usr/lib -llua5.2       -g -O0 -fsanitize=address -fno-omit-frame-pointer  
```

> Does this patch fix it?

Something is missing in your patch. Where is updating_screen defined?
Also, setting it to TRUE is done outside FEAT_GUI whereas setting to
FALSE is done inside FEAT_GUI.

**Comment by gerw** [comment]
As far as I know, I do not use concealing. For me, vim (with vim-airline) crashes with my `vimrc` 'more reliably' than with the help file. Is some concealing used in vim-airline?

The patch by @chrisbra does not help, but with @dpelle's patch I do not get any crashes.

@nuko8: Yes, this seems to be quite similar. I am also using a tiling window manager (i3wm), if that matters.

**Comment by brammool** [comment]
Dominique Pellé wrote:

Here is a complete stack with valgrind using gvim 8.0.271 gtk2:

$ valgrind --num-callers=120 ./vim -f -g -c help 2> vg.log

... and then resizing the window a couple times gives this error:

[...]

> So at regexp_nfa.c:6908, line_breakcheck() is called, and inside
> that call the shell_resized is handled, which frees some memory
> and somehow this messed up the regexp engine which later uses
> free memorry.
> 
> I'm not sure how to properly fix this, but with the following
> workaround, it never crashes:
> 
> ```
> diff --git a/src/regexp_nfa.c b/src/regexp_nfa.c
> index de22dcb..a561234 100644
> --- a/src/regexp_nfa.c
> +++ b/src/regexp_nfa.c
> @@ -6905,7 +6905,7 @@ nextchar:
>             break;
>  
>         /* Allow interrupting with CTRL-C. */
> -       line_breakcheck();
> +       /* line_breakcheck(); */
>         if (got_int)
>             break;
>  #ifdef FEAT_RELTIME
> ```
> 
> Can bug submitter also verify that this patch avoids the crash?
> 
> A better patch may be needed, but it hopefully helps to diagnose the issue.

Oh, that is quite naste.  There appear to be two problems:

1. update_single_line() does not respect or set updating_screen, the
flag that prevents recursive screen updating.  That needs to be added.

2. ui_breakcheck() causes the resize event in the GUI to be triggered.
It's supposed to not let any of that happen.  Probably setting
updating_screen in ui_breakcheck() avoids that.

-- 
TIM: To the north there lies a cave,  the cave of Caerbannog, wherein, carved
     in mystic runes, upon the very living rock, the last words of Olfin
     Bedwere of Rheged make plain the last resting place of the most Holy
     Grail.
                 "Monty Python and the Holy Grail" PYTHON (MONTY) PICTURES LTD

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///

**Comment by chrisbra** [comment]
@dpelle updating_screen is defined in globals.h and used to avoid recursively changing the screen. Not sure, why it does not fix the crash, from your stack trace, it should avoid that.

**Comment by dpelle** [comment]
@chrisbra wrote:
> updating_screen is defined in globals.h and used to avoid recursively
> changing the screen. Not sure, why it does not fix the crash, from your
> stack trace, it should avoid that.

Sorry, I now see that updating_screen a global variable indeed. I applied your
patch but it still crashes with asan, or give errors with valgrind when resizing
the screen. So same as what bug submitter also wrote above.
I have no time now to look at it, but I'll spend more time later.

Just to be sure, you installed the vim-airline plugin when trying to
reproduce it, right? For me it's very easy to reproduce the crash at least.

**Comment by vim-ml** [comment]
On Di, 31 Jan 2017, Dominique Pellé wrote:

> Just to be sure, you installed the vim-airline plugin when trying to
> reproduce it, right? For me it's very easy to reproduce the crash at least.

I am airline maintainer ;). So yes I have it installed.

Best,
Christian
-- 
Was verworren war, wird helle, was geheim, ist's fürder nicht; die
Erleuchtung wird zur Wärme, und die Wärme, sie ist Licht.
		-- Franz Grillparzer

**Comment by brammool** [comment]
Please try again after patches 8.0.0274 and 8.0.0275.  If it still crashes, please make a stack trace like Dominique made before.  There might be another code patch that causes trouble that is not checking updating_screen.

**Comment by gerw** [comment]
@brammool : With 275, the resizing is a little bit broken: The content of the gvim window does not get resized immediately after resizing the window, but only if the cursor moves to the boundary.

With 274, it is still crashing. Now, I do not have the time to produce a stack trace.

**Comment by dpelle** [comment]
For me, gvim-8.0.275 fixes the bug: no crash anymore, and resizing looks OK.

**Comment by gerw** [comment]
For me (using the tiling window manager i3), it looks sometimes like [this](https://gist.githubusercontent.com/gerw/57e69192b3a9f9cf6dfeb9bb4c8cfbdf/raw/c9f3cf60eb40065b82096e9dddfaecc4a74ea7f8/2017-01-31-230441_1600x900_scrot.png) with 275.

**Comment by gerw** [comment]
I agree that path 275 fixes the bug described in this issue. However, it introduces a new bug for me. Should I put that into a new issue?

**Comment by brammool** [comment]
Gerd Wachsmuth wrote:

> @brammool : With 275, the resizing is a little bit broken: The content of the gvim window does not get resized immediately after resizing the window, but only if the cursor moves to the boundary.
> 
> With 274, it is still crashing. Now, I do not have the time to produce a stack trace.

Looks like gui_may_resize_shell() needs to be called.  We might as well
call update_prepare() and update_finish().

-- 
LAUNCELOT: Isn't there a St. Aaaaarrrrrrggghhh's in Cornwall?
ARTHUR:    No, that's Saint Ives.
                 "Monty Python and the Holy Grail" PYTHON (MONTY) PICTURES LTD

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///

**Comment by gerw** [comment]
@brammool : If you give me a patch, I could try it.

By the way: if the window is screwed up after resizing, `:redraw` and `ctrl+L` fix it.

**Comment by gerw** [comment]
I tried patch 286, and it still does not work after resizing.

**Comment by chrisbra** [comment]
Does that only happen, when it was previously resizing during a redraw event or always now? Or in other words, does the broken resizing only happen with airline installed, in which case we would need to postpone the resizing event somehow?

**Comment by gerw** [comment]
It does only appear with vim-airline. If I disable the plugin, resizing is ok.

**Comment by brammool** [comment]
Christian Brabandt wrote:

> Does that only happen, when it was previously resizing during a redraw
> event or always now? Or in other words, does the broken resizing only
> happen with airline installed, in which case we would need to postpone
> the resizing event somehow?

We already postpone the resizing.  That is, assuming it happens as the
stacktrace shows.  When update_single_line() goes to ui_breakcheck(),
which calls gui_mch_update() which then sees the resize,
gui_resize_shell() bails out early, setting new_pixel_heigh.  When
drawing is done update_single_line() then invokes gui_may_resize_shell()
which calls gui_resize_shell() again.

Perhaps there is another sequence of events.

-- 
Dreams are free, but there's a small charge for alterations.

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///

**Comment by gerw** [comment]
Are there any news concerning the resizing issue?

**Comment by chrisbra** [comment]
wasn't that fixed by https://github.com/vim/vim/releases/tag/v8.0.0299

**Comment by gerw** [comment]
Yes, this patch fixes the problem. Thank you for your fast response!
